src/Raven.Server/ServerWide/ServerStore.cs

```diff
@@ -3271,6 +3271,12 @@ private async Task<(long Index, object Result)> SendToLeaderAsyncInternal(Transa
                     var response = await SendToNodeAsync(context, cachedLeaderTag, cmd, reachedLeader, token);
                     return (response.Index, cmd.FromRemote(response.Result));
                 }
+                catch (ConcurrencyException ex) when(reachedLeader.Value)
```

Comment:
I'm afraid this is not good enough.. 

Currently the `RachisConcurrencyException` is too generic, we need to have a finer distinguish between the various exceptions.
I'm don't think that `Conflict` is the proper status to return for this case.

Unlike the case for adding a database, where this status does make sense.

Reply:
Done